test/es6_proxy_strategy_test.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+@TestOn('chrome')
+library polymer_interop.test.convert_test;
+
+import 'dart:html';
+import 'dart:js';
+import 'package:polymer_interop/polymer_interop.dart';
+import 'package:smoke/mirrors.dart' as smoke;
+import 'package:smoke/smoke.dart';
+import 'package:test/test.dart';
+import 'package:web_components/web_components.dart';
+
+main() async {
+  smoke.useMirrors();
+  await initWebComponents();
+
+  group('es6 proxy strategy', () {
+    setUpAll(() {
+      PolymerInteropConfiguration.listConversionStrategy = checkForEs6ProxySupport();
+      PolymerInteropConfiguration.mapConversionStrategy = checkForEs6ProxySupport();
+    });
+
+    tearDownAll(() {
+      PolymerInteropConfiguration.listConversionStrategy =
+          JsInteropStrategy.mixedMode;
+      PolymerInteropConfiguration.mapConversionStrategy =
+          JsInteropStrategy.mixedMode;
+    });
+
+    group('checks',() {
+      test('es6 proxy support',() {
+        expect(checkForEs6ProxySupport(),JsInteropStrategy.es6Proxy,reason:'es6 proxy not supported');
+      });
+    });
+
+    group('list', () {
+      test('push',() {
+        List myList = [];
+        JsArray jsArray = convertToJs(myList);
+        jsArray.callMethod('push',['a']);
+        expect(myList.length,1);
+        expect(myList[0],'a');
+      });
+
+      test('slice',() {
+        List myList = ['a','b','c'];
+        JsArray jsArray = convertToJs(myList);
+
+        JsArray sliced = jsArray.callMethod('slice',[1]);
+
+        expect(sliced.length,2);
+        expect(sliced[0],'b');
+        expect(sliced[1],'c');
+
+      });
+
+      test('delete',() {
+        List myList = ['a','b','c'];
+        JsArray jsArray = convertToJs(myList);
+
+        jsArray.callMethod('delete',['1']);
+
+        expect(jsArray.length,2);
+        expect(jsArray[0],'b');
+        expect(jsArray[1],'c');
+
+      },skip:'fix this test');
+
+      test('spice',() {
+        List myList = ['a','b','c'];
+        JsArray jsArray = convertToJs(myList);
+
+        JsArray removed = jsArray.callMethod('splice',[1,1,'b1','b2']);
+
+        expect(removed.length,1);
+        expect(removed[0],'b');
+        expect(myList.length,4);
+        expect(myList[1],'b1');
+        expect(myList[2],'b2');
+        expect(myList[3],'c');
+
+      });
+
+
+      test('in and out', () {
+        List myList = [];
+        JsObject x = convertToJs(myList);
+        x.callMethod("push", ['val1']);
+        expect(myList.length, 1);
+        expect(myList[0], 'val1');
+
+        myList.add('val2');
+        expect(x['length'], 2);
+        expect(x[1], 'val2');
+
+        List otherList = convertToDart(x);
+
+        expect(otherList == myList, isTrue);
```

Comment:
I think this should this be an `identical(otherList, myList)` check, just to be sure? Same for other tests.